### ISSUE TITLE
Add subcommand to download Emscripten

### ISSUE THREAD

**Issue by ndarilek** [koute/cargo-web#22]
I'd like to download Emscripten, then run commands in the downloaded installation to install ports. This is for a CI environment generating automated builds.

I don't see a way to do this other than triggering a build, which will fail since the environment lacks the necessary tools, then using the freshly-installed Emscripten to set up the environment for a second build that succeeds. A `download` command would help with that. Or maybe an `emscripten` command so other subcommands might be added in the future. I don't care much about what the command is called, as long as I can just download whatever Emscripten version cargo-web supports so I can install ports before building. :)

Thanks.

**Comment by koute** [comment]
Would this work for you?

```
$ git clone https://github.com/Rust-SDL2/rust-sdl2.git
$ cd rust-sdl2
$ echo 'link-args = ["-s", "USE_SDL=2"]' > Web.toml
$ cargo web build --example game-of-life
# It builds without errors!
```

Basically, if you create a `Web.toml` in your crate you can put link args for Emscripten in it, and [according to Emscripten docs](https://kripken.github.io/emscripten-site/docs/compiling/Building-Projects.html#emscripten-ports) it will build the port for you automatically.

**Comment by ndarilek** [comment]
Hmm, I suppose that would work, thanks! I'm new enough to all of this 
that I'm not entirely sure whether I might have some *other* need to 
only download the tooling, but if this is true of all ports then my use 
case is addressed.

Maybe installing non-port libraries might require emcc access? In which 
case, I'd want cargo-web to do a separate download step so I can build 
my libraries.

**Comment by koute** [comment]
Well, I certainly wouldn't be against adding such a command, perhaps something like `cargo web prepare-emscripten` which would output `export EMSCRIPTEN=<...>\nexport PATH=<...>` etc. so you could do `$(cargo web prepare-emscripten)` and have access to Emscripten in your shell session.

**Comment by ndarilek** [comment]
`prepare`, I like that. Right now it isn't immediately obvious that 
cargo-web significantly simplifies Emscripten setup, at least for the 
case where I'm only using it with Rust. Having a command that makes this 
a bit more obvious would be nice.

**Comment by koute** [comment]
If you'd like to try implementing it ~~and you don't mind having to deal with a bunch of code that definitely needs some refactoring~~ I'd welcome a pull request. (:

It should be *relatively* simple to add, I think.

**Comment by svenstaro** [comment]
Fixed by #52 so this can be closed.
